src/status_im2/contexts/communities/menus/channel_options/view.cljs

```diff
@@ -0,0 +1,109 @@
+
+(defn unmute-channel-action
+  [chat-id]
+  (hide-sheet-and-dispatch [:chat.ui/mute chat-id false]))
+
+(defn view-members
+  [id]
+  {:icon                :i/members
+   :accessibility-label :view-members
+   :label               (i18n/label :t/view-channel-members-and-details)})
+
+(defn view-token-gating
+  [id]
+  {:icon                :i/token
+   :right-icon          :i/chevron-right
+   :chevron?            true
+   :accessibility-label :view-token-gating
+   :on-press            #(js/alert (str "implement action" id))
+   :label               (i18n/label :t/view-token-gating)})
+
+(defn mark-as-read
+  [id]
+  {:icon                :i/up-to-date
+   :accessibility-label :mark-as-read
+   :label               (i18n/label :t/mark-as-read)
+   :on-press            #()})
+
+(defn mute-channel
+  [id muted?]
+  {:icon                (if muted? :i/muted :i/activity-center)
+   :accessibility-label (if muted? :unmute-channel :mute-channel)
+   :label               (i18n/label (if muted? :t/unmute-channel :t/mute-channel))
+   :on-press            (if muted?
+                          #(unmute-channel-action id)
+                          #(mute-channel-action id))
+   :right-icon          :i/chevron-right
+   :chevron?            true})
+
+(defn fetch-messages
+  []
+  {:icon                :i/save
+   :label               (i18n/label :t/fetch-messages)
+   :on-press            #(js/alert "TODO: to be implemented, requires design input")
+   :danger?             false
+   :accessibility-label :fetch-messages
+   :sub-label           nil
+   :right-icon          :i/chevron-right
+   :chevron?            true})
+
+(defn invite-contacts
+  [id]
+  {:icon                :i/add-user
+   :accessibility-label :invite-people-from-contacts
+   :label               (i18n/label :t/invite-people-from-contacts)
+   :on-press            #(hide-sheet-and-dispatch [:communities/invite-people-pressed id])})
+
+(defn show-qr
+  [id]
+  {:icon                :i/qr-code
+   :accessibility-label :show-qr
+   :on-press            #(js/alert (str "implement action" id))
+   :label               (i18n/label :t/show-qr)})
+
+(defn share-channel
+  [id]
+  {:icon                :i/share
+   :accessibility-label :share-channel
+   :label               (i18n/label :t/share-channel)
+   :on-press            #()})
+
+(defn notifications
+  []
+  {:icon                :i/notifications
+   :label               (i18n/label :t/notifications)
+   :on-press            #(js/alert "TODO: to be implemented, requires design input")
+   :danger?             false
+   :sub-label           (i18n/label :t/all-messages)
+   :accessibility-label :manage-notifications
+   :right-icon          :i/chevron-right
+   :chevron?            true
+   :add-divider?        false})
+
+(defn channel-options-bottom-sheet
+  [community-id id]
+  (let [{:keys [token-gated?]} (rf/sub [:communities/community id])
+        {:keys [muted]}        (rf/sub [:chat-by-id (str community-id id)])]
```

Comment:
I see a bunch of places doing this ` (str community-id id)` to create a composite key. This is a bit error prone, easy to miss I'd say, so perhaps something to consider in a future refactor. One idea is we could enrich the community chat map with the composite key at "write app db time" (when the data arrives), for instance using a *data store* namespace. This way no other place of the code needs to worry about concatenating them again.

I thought I'd also share something just in case. Clojure map keys can be almost anything, whereas Javascript keys must be strings or symbols. Concatenating things to build up a composite key is very Javascript-esque. A more idiomatic approach in Clojure is to represent the key as a tuple, for example `[community-id id]`. This also has the added advantage of being human readable if we inspect the app db, and the individual IDs can be easily extracted from keys because it's just a data structure. Performance wise it should be just as good if the individual IDs are strings, because hashing the tuple would be fairly quick, especially at the order of magnitude of data we're dealing with in mobile. So avoiding this idiomatic approach due to performance considerations could be leaning on the side of micro optimization.

Just a thought (not a suggestion for change here), of course messing around with the app db is no joke because it's easy to introduce bugs without a good test suite.

@cammellos, care to share a quick opinion :relaxed: ?

Reply:
> Doing str .. in a single place is definitely a good suggestion, we can do that in status-go if the client wants this, so that clojure just has to rename the key

:+1: That would be even better so we can keep the client's knowledge of what an ID is more opaque.